pkg/kubelet/cm/dra/manager.go

```diff
@@ -495,6 +509,51 @@ func (m *Manager) GetResources(pod *v1.Pod, container *v1.Container) (*Container
 			}
 		}
 	}
+	if !utilfeature.DefaultFeatureGate.Enabled(kubefeatures.DRAExtendedResource) {
+		return &ContainerInfo{CDIDevices: cdiDevices}, nil
+	}
+	if pod.Status.ExtendedResourceClaimStatus != nil {
+		claimName := &pod.Status.ExtendedResourceClaimStatus.ResourceClaimName
+		// if the container has requests for extended resources backed by DRA,
+		// they must have been allocated via the extendedResourceClaim created
+		// by the kube-scheduler.
+		err := m.cache.withRLock(func() error {
+			claimInfo, exists := m.cache.get(*claimName, pod.Namespace)
+			if !exists {
+				return fmt.Errorf("unable to get claim info for claim %s in namespace %s", *claimName, pod.Namespace)
+			}
+
+			// As of Kubernetes 1.31, CDI device IDs are not passed via annotations anymore.
+			for rName, rValue := range container.Resources.Requests {
+				if rValue.IsZero() {
+					// We only care about the resources requested by the pod
+					continue
+				}
+				switch rName {
+				case v1.ResourceCPU:
+				case v1.ResourceMemory:
+				case v1.ResourceEphemeralStorage:
+				default:
+					if v1helper.IsExtendedResourceName(rName) {
+						requestName := ""
+						for _, rm := range pod.Status.ExtendedResourceClaimStatus.RequestMappings {
+							if rm.ContainerName == container.Name && rm.ResourceName == rName.String() {
+								requestName = rm.RequestName
+								break
+							}
+						}
+						if requestName != "" {
+							cdiDevices = append(cdiDevices, claimInfo.cdiDevicesAsList(requestName)...)
+						}
+					}
+				}
```

Comment:
Would it be clearer to just use `if v1helper.IsExtendedResourceName(rName) {` here instead of the switch?